reconcile/utils/ocm/upgrades.py

```diff
@@ -37,13 +37,29 @@ def get_addon_upgrade_policies(
     ):
         if addon_id and policy["addon_id"] != addon_id:
             continue
-        results.append(
-            {k: v for k, v in policy.items() if k in ADDON_UPGRADE_POLICY_DESIRED_KEYS}
+        policy_data = {
+            k: v for k, v in policy.items() if k in ADDON_UPGRADE_POLICY_DESIRED_KEYS
+        }
+        policy_data["state"] = get_addon_upgrade_policy_state(
```

Comment:
won't that do too many calls to the OCM API?

Reply:
there is no other way to get the state. at least for HCP the state is embedded in the upgradepolicy.

i ack the problem but in the end it does not matter too much for now, because not too many upgrade policies exist at the same time.